### ISSUE TITLE
`delete --clean` deletes active plugins if run from a script

### ISSUE THREAD

**Issue by GaryBoone** [zdharma/zinit#384]
**Describe the bug**
When run from a script, the plugins aren't loaded. As a result, `zi delete --clean` deletes the unloaded plugins which causes them all to be redownloaded when Zinit is next restarted.

**To Reproduce**
Make a script containing:
```
#!/usr/local/bin/zsh. # Or whenever zsh is located.
zinit self-update
zinit update
zi delete --clean --yes
```

1. Run this script. The currently loaded plugins will be deleted with a message like: "Deleting the following XX UNLOADED plugins:" 

2. `$ exec zsh` to restart the shell and Zinit. All of the plugins will be redownloaded and initialized.

It doesn't help to attempt to load the plugins in the script by adding a line like:

   `source "$HOME/.zshrc"`

**Expected behavior**
When run from a script, `delete --clean` should not delete loaded plugins.

**Versions:**
 - $ZSH_VERSION == 5.8
 - Operating system name and version (uname -a): Darwin Kernel Version 19.5.0

**Additional context**
The purpose of the script above is to create an automatic script that updates zinit and its snippets/plugins. This script would include other updates like `brew update`.

While it might work to source the commands (`source update_script.sh`) so they run in the current shell, that won't work if those commands are included in another script that does start its own shell.

**Comment by pschmitt** [comment]
You could just source your config without any async loading.

Or make it a function. That'll work too.

**Comment by GaryBoone** [comment]
Thanks, @pschmitt . Could you say more? How do you source your config without async loading? Make which a function? How do you make it a function?

**Comment by GaryBoone** [comment]
Oh, I get it. I just made the update shell script into a zsh function. To be explicit, I created a file called updateAll.zsh, which contained

```
updateAll () {
  echo "*** updating zinit and packages  ***"
  zinit self-update
  zinit update
  zi delete --clean --yes
}
```

Then added `source updateAll.zsh` to my ~/.zshrc. 

So now in the terminal, I just type `$ updateAll` to update everything. 

The result of making it a function is that the commands run in the current shell, so the plugins are all loaded. Therefore, there are none seen as unloaded and then deleted by the `zi delete` command.
